include/Gaffer/ThreadState.h

```diff
@@ -121,7 +122,26 @@ class GAFFER_API ThreadState
 			return m_context;
 		}
 
+		enum class CacheType
+		{
+			Value,
+			Hash
+		};
```

Comment:
`Process::processType()` is deliberately an `InternedString` rather than an enum to allow it to be an open-ended system : extension modules can introduce their own process types (and have them be monitored) without modifying the core. One example of this is the [various process types](https://github.com/GafferHQ/gaffer/blob/main/src/GafferDispatch/TaskNode.cpp#L91-L123) used by GafferDispatch. To date, none of ThreadState, Monitor or Process have needed any concept of what valid process types might be (or even that ValuePlug exists), and I'd like to keep it that way. That's why I'm advocating for passing `InternedString processType` rather than this new `CacheType` enum.